modules/server/src/main/scala/higherkindness/compendium/db/PgDBService.scala

```diff
@@ -22,18 +22,22 @@ import doobie.util.transactor.Transactor
 import doobie.implicits._
 import higherkindness.compendium.core.refinements.ProtocolId
 import higherkindness.compendium.db.queries.Queries
+import higherkindness.compendium.models.{IdlNames, MetaProtocolDB}
 
 object PgDBService {
 
   def impl[F[_]: Async](xa: Transactor[F]): DBService[F] =
     new DBService[F] {
 
-      override def upsertProtocol(id: ProtocolId): F[Unit] =
-        Queries.upsertProtocolIdQ(id.value).run.void.transact(xa)
+      override def upsertProtocol(id: ProtocolId, idlName: IdlNames): F[Unit] =
+        Queries.upsertProtocolIdQ(id.value, idlName.entryName).run.void.transact(xa)
 
       override def existsProtocol(id: ProtocolId): F[Boolean] =
         Queries.checkIfExistsQ(id.value).unique.transact(xa)
 
+      override def selectProtocolBytId(id: ProtocolId): F[MetaProtocolDB] =
```

Comment:
Typo?

```suggestion
      override def selectProtocolById(id: ProtocolId): F[MetaProtocolDB] =
```

Reply:
I'm gonna do this change locally since I have to perform others